gcp_variant_transforms/libs/header_merger_test.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2019 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test cases for header_merger module."""
+
+from collections import OrderedDict
+import unittest
+import vcf
+
+from gcp_variant_transforms.beam_io import vcf_header_io
+from gcp_variant_transforms.libs import vcf_field_conflict_resolver
+from gcp_variant_transforms.libs.header_merger import HeaderMerger
+
+FILE_1_LINES = [
+    '##fileformat=VCFv4.2\n',
+    '##INFO=<ID=NS,Number=1,Type=Integer,Description="Number samples">\n',
+    '##INFO=<ID=AF,Number=A,Type=Float,Description="Allele Frequency">\n',
+    '##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">\n',
+    '##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="GQ">\n',
+    '#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	Sample1	Sample2\n']
+FILE_2_LINES = [
+    '##fileformat=VCFv4.2\n',
+    '##INFO=<ID=NS2,Number=1,Type=Integer,Description="Number samples">\n',
+    '##INFO=<ID=AF,Number=A,Type=Float,Description="Allele Frequency">\n',
+    '##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">\n',
+    '##FORMAT=<ID=GQ2,Number=1,Type=Integer,Description="GQ">\n',
+    '#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	Sample3\n']
+
+class HeaderMergerTest(unittest.TestCase):
+  """Test cases for GetMergeHeaders `PTransform`."""
+
+  def _get_header_from_reader(self, reader):
+    """Extracts values from a pyVCF reader into a VcfHeader object."""
+    return vcf_header_io.VcfHeader(
+        infos=reader.infos,
+        filters=reader.filters,
+        alts=reader.alts,
+        formats=reader.formats,
+        contigs=reader.contigs)
+
+  def _get_header_merger(self, split_alternate_allele_info_fields=True):
+    resolver = vcf_field_conflict_resolver.FieldConflictResolver(
+        split_alternate_allele_info_fields)
+    merger = HeaderMerger(resolver)
+    return merger
+
+  def test_merge_header_with_empty_one(self):
+    vcf_reader = vcf.Reader(fsock=iter(FILE_1_LINES))
+    header = self._get_header_from_reader(vcf_reader)
+    merger = self._get_header_merger()
+    empty_header = vcf_header_io.VcfHeader()
+
+    merged_header_1 = header
```

Comment:
nit: I think it would be easier to read by removing `merged_header_1` and `merged_header_2`. Same for some of the other test cases.

Reply:
Ummm, done everywhere except for here :D 

Modified all the vars to main_header (the one in which the merge is going to happen) and to secondary (the one that is being added to the first one). This specific test case is slightly different as it tests 2 cases - merging empty into non empty header and merging non empty into empty header (hence two main header vars).

Reply:
I am sorry I didn't make it clear. I mean we can simplify the code like:
`header_1 = self._get_header_from_reader(vcf_reader)`
`header_2 = vcf_header_io.VcfHeader()`
`merger.merge(header_1, header_2)`
`merger.merge(header_2, header_1)`
WDYT?

Reply:
Makes sense. Done.